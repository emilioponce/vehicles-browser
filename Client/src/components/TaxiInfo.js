import React from 'react'
import PropTypes from 'prop-types'

import styles from './TaxiInfo.module.css'

const TaxiInfo = ({ title, position, state }) => {
  return (
    <div className={styles.taxiInfo}>
      <div className={styles.title}>{title}</div>
      <div>Position:</div>
      <div className={styles.coordinates}>
        <div>Latitude: {position.latitude.toFixed(4)}</div>
        <div>Longitude: {position.longitude.toFixed(4)}</div>
      </div>
      <span
        className={
          state === 'ACTIVE' ? styles.activeState : styles.inactiveState
        }
      >
        {state}
      </span>
    </div>
  )
}

TaxiInfo.propTypes = {
  title: PropTypes.string.isRequired,
  position: PropTypes.object.isRequired,
  state: PropTypes.string.isRequired
}

export default TaxiInfo
