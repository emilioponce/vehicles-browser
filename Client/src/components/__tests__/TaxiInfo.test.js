import React from 'react'
import ReactDOM from 'react-dom'
import TaxiInfo from '../TaxiInfo'
import renderer from 'react-test-renderer'

const component = (
  <TaxiInfo
    position={{
      latitude: 41.5532,
      longitude: 2.10088
    }}
    state="ACTIVE"
    title="TAXI"
  />
)
describe('Component TaxiInfo', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('snapshot renders correctly', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
