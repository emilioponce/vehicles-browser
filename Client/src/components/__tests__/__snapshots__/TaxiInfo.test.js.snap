// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TaxiInfo snapshot renders correctly 1`] = `
<div
  className="taxiInfo"
>
  <div
    className="title"
  >
    TAXI
  </div>
  <div>
    Position:
  </div>
  <div
    className="coordinates"
  >
    <div>
      Latitude: 
      41.5532
    </div>
    <div>
      Longitude: 
      2.1009
    </div>
  </div>
  <span
    className="activeState"
  >
    ACTIVE
  </span>
</div>
`;
